tpm/tpm.go

```diff
@@ -184,25 +273,29 @@ func (t *TPM) close(ctx context.Context) error {
 
 	// clean up the attest.TPM
 	if t.attestTPM != nil {
-		err := t.attestTPM.Close()
-		t.attestTPM = nil
-		if err != nil {
+		defer func() { t.attestTPM = nil }()
+		if err := closer.AttestTPM(t.attestTPM, t.attestConfig); err != nil {
 			return fmt.Errorf("failed closing attest.TPM: %w", err)
 		}
 	}
 
 	// clean up the go-tpm rwc
 	if t.rwc != nil {
-		err := t.rwc.Close()
-		t.rwc = nil
-		if err != nil {
+		defer func() { t.rwc = nil }()
+		if err := closer.RWC(t.rwc); err != nil {
```

Comment:
It might be ok, but it might be weird to set those to nil even if the close fails.

Reply:
Yea, I kind of agree. But I think it's OK in this case, because no matter what happens, I want to have a new, clean `io.RWC` on the next `open`, and never start with an old one in a weird state.